lib/src/layer/tile_layer/tile_manager.dart

```diff
@@ -0,0 +1,264 @@
+import 'package:flutter_map/flutter_map.dart';
+import 'package:flutter_map/src/core/bounds.dart';
+import 'package:flutter_map/src/layer/tile_layer/coords.dart';
+import 'package:tuple/tuple.dart';
+
+class TileManager {
+  final Map<String, Tile> _tiles = {};
+
+  void abortLoading(double? tileZoom, EvictErrorTileStrategy evictionStrategy) {
+    final toRemove = <String>[];
+    for (final entry in _tiles.entries) {
+      final tile = entry.value;
+
+      if (tile.coords.z != tileZoom) {
+        if (tile.loaded == null) {
+          toRemove.add(entry.key);
+        }
+      }
```

Comment:
Using nested if statements is generally worse for performance than a logic gate, especially where readability will not be affected that much.

```suggestion
      if (tile.coords.z != tileZoom && tile.loaded == null) {
        toRemove.add(entry.key);
      }
```